generator/internal/sidekick/update.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sidekick
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"maps"
+	"net/http"
+	"os"
+	"strings"
+
+	toml "github.com/pelletier/go-toml/v2"
+)
+
+const (
+	defaultGitHubApi = "https://api.github.com"
+	defaultGitHub    = "https://github.com"
+	repo             = "googleapis/googleapis"
+	branch           = "master"
+)
+
+func update(rootConfig *Config, cmdLine *CommandLine) error {
+	if err := updateRootConfig(rootConfig); err != nil {
+		return err
+	}
+	// Reload the freshly minted configuration.
+	rootConfig, err := loadRootConfig(".sidekick.toml")
+	if err != nil {
+		return err
+	}
+	return refreshAll(rootConfig, cmdLine)
+}
+
+func updateRootConfig(rootConfig *Config) error {
+	gitHubApi, ok := rootConfig.Source["github-api"]
+	if !ok {
+		gitHubApi = defaultGitHubApi
+	}
+	gitHub, ok := rootConfig.Source["github"]
```

Comment:
Note: I think this is fine for now. But in the future we may want to also honor an environment variable to source a local github token if we notice we are running into quota issues. There are some good libs out there that make all this pretty easy that I have used before.